sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java

```diff
@@ -122,21 +123,26 @@ protected void scanFile(PythonInputFile inputFile) throws IOException {
     } catch (RecognitionException e) {
       visitorContext = new PythonVisitorContext(pythonFile, e, context.runtime().getProduct());
 
-      var line = (inputFile.kind() == PythonInputFile.Kind.IPYTHON) ? ((GeneratedIPythonFile) inputFile).locationMap().get(e.getLine()).line() : e.getLine();
+      var line = (inputFile.kind() == PythonInputFile.Kind.IPYTHON) ?
+        ((GeneratedIPythonFile) inputFile).locationMap().get(e.getLine()).line() : e.getLine();
       var newMessage = e.getMessage().replace("line " + e.getLine(), "line " + line);
 
-      LOG.error("Unable to parse file: " + inputFile);
+      LOG.error("Unable to parse file: {}", inputFile);
       LOG.error(newMessage);
-      context.newAnalysisError()
+
+      NewAnalysisError analysisError = context.newAnalysisError()
         .onFile(inputFile.wrappedFile())
-        .at(inputFile.wrappedFile().newPointer(line, 0))
-        .message(newMessage)
-        .save();
+        .message(newMessage);
+
+      if (line > 0) {
```

Comment:
If a lexer error is caught, the exception is wrapped in a `RecognitionException` with the `line` set to `0`. Subsequently, this will cause an `IllegalStateException` in the `at` method below.

That being said, I'm not sure if this is 100% kosher...

Reply:
Also, this line is currently not really tested. I'm not sure if it makes sense to test it.

Reply:
The issue I see is with the Notebooks, I am not sure that in the JSON file the first line we refer to is 1. We should be careful about that.